File: Examples2018/DataManagementSQL/app/src/main/java/course/examples/datamanagement/sql/DatabaseOpenHelper.java
package course.examples.datamanagement.sql;

import android.content.Context;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;

class DatabaseOpenHelper extends SQLiteOpenHelper {
	
	final static String TABLE_NAME = "artists";
	final static String ARTIST_NAME = "name";
	private final static String _ID = "_id";
	final static String[] columns = { _ID, ARTIST_NAME };

	final private static String CREATE_CMD =

	"CREATE TABLE artists (" + _ID
			+ " INTEGER PRIMARY KEY AUTOINCREMENT, "
			+ ARTIST_NAME + " TEXT NOT NULL)";

	final private static String NAME = "artist_db";
	final private static Integer VERSION = 1;
	final private Context mContext;

	public DatabaseOpenHelper(Context context) {
		super(context, NAME, null, VERSION);
		this.mContext = context;
	}

	@Override
	public void onCreate(SQLiteDatabase db) {
		db.execSQL(CREATE_CMD);
	}

	@Override
	public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
		// N/A
	}

	void deleteDatabase() {
		mContext.deleteDatabase(NAME);
	}
}

File: ExamplesKotlin/DataManagementSQL/app/src/main/java/course/examples/datamanagement/sql/DatabaseOpenHelper.kt
package course.examples.datamanagement.sql

import android.content.Context
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper

internal class DatabaseOpenHelper(private val mContext: Context) :


    SQLiteOpenHelper(mContext, NAME, null, VERSION) {

    companion object {
        const val TABLE_NAME = "artists"
        const val ARTIST_NAME = "name"
        private const val ID = "_id"
        val columns = arrayOf(ID, ARTIST_NAME)
        const val CREATE_CMD = (
                "CREATE TABLE artists (" + ID
                        + " INTEGER PRIMARY KEY AUTOINCREMENT, "
                        + ARTIST_NAME + " TEXT NOT NULL)")
        const val NAME = "artist_db"
        const val VERSION = 1
    }

    override fun onCreate(db: SQLiteDatabase) {
        db.execSQL(CREATE_CMD)
    }

    override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {
        // N/A
    }

    fun deleteDatabase() {
        mContext.deleteDatabase(NAME)
    }
}
